MapleServer2/Packets/Helpers/ItemPacketHelper.cs

```diff
@@ -80,6 +93,15 @@ public static PacketWriter WriteItem(this PacketWriter pWriter, Item item)
 
         private static PacketWriter WriteAppearance(this PacketWriter pWriter, Item item)
         {
+            // TODO: make WriteAppearance proper for GEM, this is hacky.
```

Comment:
Just make a `WriteGemAppearance` method, and then do the below check at the calling method, if gem, write gem, if not, write normal appearance. It's not perfect but much less hacky and more readable

Reply:
Personally disagree with this as appearance is one method in the client that matches every item. It's simply not explored yet so the method is partially incorrect. Adding multiple decodes that differ from one another only slightly due to them being for a specific item we don't understand a check for results in redundant code imo.